File: app/src/main/java/com/senerunosoft/puantablosu/IGameService.java
package com.senerunosoft.puantablosu;

import com.senerunosoft.puantablosu.model.Game;
import com.senerunosoft.puantablosu.model.SingleScore;

import java.util.List;

public interface IGameService {

    Game createGame(String gameTitle);
    void addPlayer(Game game, String playerName);
    boolean addScore(Game game, List<SingleScore> scoreList);
    int getPlayerRoundScore(Game game, String playerId, int round);
    List<SingleScore> getCalculatedScore(Game game);
    String serializeGame(Game game);
    Game deserializeGame(String gameString);
}

File: app/src/main/java/com/senerunosoft/puantablosu/IGameService.kt
package com.senerunosoft.puantablosu

import com.senerunosoft.puantablosu.model.Game
import com.senerunosoft.puantablosu.model.SingleScore
import com.senerunosoft.puantablosu.model.config.IConfig
import com.senerunosoft.puantablosu.model.enums.GameType

interface IGameService {
    fun createGame(gameTitle: String, gameType: GameType, config: IConfig?): Game?
    fun addPlayer(game: Game?, playerName: String)
    fun addScore(game: Game?, scoreList: List<SingleScore>?): Boolean
    fun getPlayerRoundScore(game: Game, playerId: String, round: Int): Int
    fun getCalculatedScore(game: Game): List<SingleScore>
    fun serializeGame(game: Game?): String?
    fun deserializeGame(gameString: String): Game?
}